sdk/tests/conformance2/textures/misc/tex-3d-size-limit.html

```diff
@@ -80,6 +73,39 @@
         wtu.glErrorShouldBe(gl, gl.INVALID_VALUE, "texImage3D should fail for dimension out of range.");
     }
 
+    // Probe to discover the max non-OOM level.
+    var maxTestLevels = maxLevels;
+    while (true) {
+        gl.texImage3D(gl.TEXTURE_3D, maxTestLevels-1, gl.RGBA, 1, 1, 1, 0, gl.RGBA, gl.UNSIGNED_BYTE, null);
+        var err = gl.getError();
+        if (err == gl.OUT_OF_MEMORY) {
+            debug("Probe failed for level=" + (maxTestLevels-1) + ", reducing...");
+            maxTestLevels -= 1;
+            if (!maxTestLevels) {
+                testFailed("Failed to allocate any levels");
+                return;
+            }
```

Comment:
Taking that the test fails here don't you think it is better to just break it out of the loop here itself and make it go to clean up? 
if I am missing something and it is testing multiple things should it not be either in different tests or enumerate test cases?

Reply:
@jaibeer72 the test has to find the maximum mipmap level that can be allocated in order for the test loop below to work. It's appropriate for the test to fail if discovery of the maximum supported mipmap level fails.